src/core/main.cpp

```diff
@@ -99,6 +99,16 @@ void signalHandler( int signum ) {
 }
 #endif
 
+void interruptHandler(int code) {
+	using namespace lmms::gui;
+
+	if(getGUI() != nullptr) {
+		getGUI()->sendInterrupt(code);
+	} else {
+		exit(3);
+	}
+}
+
```

Comment:
I considered removing this and binding `signal(SIGINT` directly to the GuiApplication function, but I wanted to also capture the interrupt from CLI and I thought this was the cleanest way to handle both.

Reply:
I'm no longer catching the SIGINT in this fashion, which does leave the terminal abandoned.  Since unix signal handlers are so damn strict, quoting:

> It's not safe to call a non-async safe function from within the signal handler

... then we can't even check to see if we have initialized `lmms::gui::getGUI()`, we can only write to an arbitrary file handle and hope it's received on the other end.    Resolving this for now, but I wouldn't be surprised if this PR is failed for this reason alone.